### PR TITLE
[Proof of concept] Drop mixin pattern for ContainerRuntime

### PR DESCRIPTION
**Pull request by markfields**
## Description

We have used the mixin pattern to inject additional behavior into ContainerRuntime, most notably for adding support for Attribution.

This pattern is powerful because it allows arbitrary composition - each `loadRuntime` function takes in a base constructor, which could be an already-mixed in version.

The downside is that it requires access to the ContainerRuntime class itself.  We have just deprecated this class as an export from our packages; it's only to be used internally within FF now.

What this PR does is replace the mixin pattern for adding support for Attribution with a one-off function `loadRuntimeWithAttribution` that returns an instance of ContainerRuntime with the additional behavior.  (It adds a dataStore which it delegates that responsibility to).

We should consider if this pattern can be adjusted to allow composition. Our goal is to simplify the ContainerRuntime class and having this capability implemented in this simpler way could be advantageous.

## Reviewer Guidance

This is just to spark discussion.  Not intending to merge in its current state.

### CODE REVIEW

**Comment by markfields** [comment]
Some excerpts from an internal discussion about this.  Like, why I'm thinking about it and what we need to do if anything:

* We are no longer exporting the ContainerRuntime class, so any scheme (e.g. mixin pattern) that uses `typeof ContainerRuntime` for extensibility is now limited to internal use only.
* If we want to expose a similar capability to some consumers (to add pre- and post- loading steps), we might want to put some thought into the exact pattern/API
* About composability
  * Mixin pattern lets someone mixin their functionality onto a CR without having to know whether it's got other stuff mixed in already or not.
  * We should take this into consideration for the solution. My first draft didn't support this, so I was thinking out loud about that.

_tl;dr - I wanted to raise the point that with my recent change to stop exporting ContainerRuntime, we've lost an extensibility point (that no one is using at the moment) of the encapsulated model_

For now, external users can hard-code an extension (pre- and post- loading steps), and that's good enough -- this PR demonstrates this technique. But it doesn't support composability (i.e. adding functionality without knowing what else is already there). If a generic composability requirement arises we can revisit.
